swap_meet/vendor.py

```diff
@@ -0,0 +1,201 @@
+
+        Returns:
+            bool: If this `Vendor`'s `inventory` doesn't contain `my_item` or
+                the `friend`'s `inventory` doesn't contain `their_item`,
+                returns `False`; otherwise, returns `True`.
+        """
+
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.remove(my_item)
+            friend.add(my_item)
+            friend.remove(their_item)
+            self.add(their_item)
+            return True
+        return False
+
+    def swap_first_item(self, friend):
+        """Swaps the first item in this `Vendor`'s `inventory` with the first item in the `friend`'s `inventory`.
+
+        Args:
+            friend (object): An instance of another `Vendor`, representing the
+                friend that this `Vendor` instance is swapping with.
+
+        Returns:
+            bool: If either itself or the `friend` have an empty `inventory`,
+                returns `False`. Otherwise, returns True.
+        """
+
+        if not self.inventory or not friend.inventory:
+            return False
+        else:
+            my_item = self.inventory.pop(0)
+            self.add(friend.inventory.pop(0))
+            friend.add(my_item)
+            return True
+
+    def get_best_by_category(self, item_category):
+        """Gets the item with the best `condition` in a certain `category`.
+
+        Args:
+            item_category (str): Represents the `category` of the items.
+
+        Returns:
+            best_item: A single item, even if there are duplicates (two or more
+                of the same item with the same `condition`). If there are no
+                items in the `inventory` that match the `category`, returns `None`.
+        """
+
+        best_cond = 0
+        best_item = None
+
+        for item in self.get_by_category(item_category):
+            if item.condition > best_cond:
+                best_item = item
+                best_cond = item.condition
+        return best_item
+
+    def swap_best_by_category(self, other, my_priority, their_priority):
+        """Swaps the best item of certain categories with another `Vendor`.
+
+        Args:
+            other (object): Represents another `Vendor` instance to trade with.
+
+            my_priority (str): Represents a `category` that this `Vendor`
+                instance wants to receive.
+
+            their_priority (str): Represents a `category` that `other` wants
+                to receive.
+
+        Returns:
+            bool: If any of both `Vendor` instances doesn't have an item that
+                matches a desired `category`, returns `False`; otherwise,
+                returns True.
+        """
+
+        mine = self.get_best_by_category(their_priority)
+        theirs = other.get_best_by_category(my_priority)
+        if not mine or not theirs:
+            return False
+        else:
+            return self.swap_items(other, mine, theirs)
+
+    def get_by_newest(self):
```

Comment:
Love that you were able to do this optional and that you created this helper method rather than trying to do all the work in `swap_by_newest`!

Reply:
glad I managed to DRY this part properly!